pycalphad/io/cs_dat.py

```diff
@@ -1159,6 +1162,999 @@ def parse_cs_dat(instring):
+        else:
+            coeff_string = '0.' + coeff_string[1] + coeff_string[3:] + ' '
+    elif abs(coeff) < 1:
+        coeff_string = f'{coeff: .16f}'[:10] + '     '
+        if coeff < 0:
+            coeff_string = ' -' + coeff_string[2:]
+    else:
+        coeff_string = f'{coeff: .16f}'[:10] + '     '
+
+    return coeff_string
+
+def format_coefficient_mag(coeff):
+    # The formatting is inconsistent, so unfortunately each value range is custom
+    if   coeff == 0:
+        coeff_string = '0.000000     '
+    elif abs(coeff) < 0.1 or abs(coeff) >= 1e6:
+        coeff_string = f'{coeff*10: .5E}'[:13]
+        if coeff < 0:
+            coeff_string = '-.' + coeff_string[1] + coeff_string[3:] + ' '
+        else:
+            coeff_string = '0.' + coeff_string[1] + coeff_string[3:] + ' '
+    elif abs(coeff) < 1:
+        coeff_string = f'{coeff: .16f}'[:8] + '     '
+        if coeff < 0:
+            coeff_string = ' -' + coeff_string[2:]
+    else:
+        coeff_string = f'{coeff: .16f}'[:8] + '     '
+
+    return coeff_string
+
+def make_constituent_mapping(constituents, endmember_params):
+    # Match endmembers to constituents they are composed of
+    constituent_mapping = [[] for _ in range(len(constituents))]
+    for endmember in endmember_params:
+        sublattice = 0
+        for speciesList in endmember['constituent_array']:
+            for species in speciesList:
+                constituent_mapping[sublattice].append(constituents[sublattice].index(species.name) + 1)
+                sublattice += 1
+    return constituent_mapping
+
+def iterative_substitution(param,symbols):
+    # Iteratively subsitutes a list of symbols into an equation
+    # Doesn't check for circular dependence
+    subs_param = param
+    requires_substitution = True
+    while requires_substitution:
+        requires_substitution = False
+        param_symbols = [s.name for s in subs_param.free_symbols]
+        # Check if any of the symbols in symbols are still in the parameter
+        common_symbols = [s for s in symbols if s in param_symbols]
+        if common_symbols:
+            # Do another round of substitutions if so
+            subs_param = piecewise_fold(simplify(subs_param.subs(symbols)))
+            requires_substitution = True
+    return subs_param
+
+def simplify_reference_gibbs(equation,symbols):
+    # Determine equation type and number of intervals
+    # Is this monstrosity necessary? Yes, it would seem so.
+    simplified_parameter = iterative_substitution(equation,symbols)
+    gibbs_equation = expand(simplify(simplify(simplified_parameter))).args
+    if not gibbs_equation:
+        # If all ranges have 0 value, simplify returns empty set, so revert
+        simplified_parameter = equation.args
+        gibbs_equation = []
+        for i in range(int(len(simplified_parameter)/2)):
+            gibbs_equation.append((simplified_parameter[i*2],simplified_parameter[i*2+1]))
+    return gibbs_equation
```

Comment:
There's probably another way to accomplish the goal here, but I don't fully understand the purpose of this function yet. I'd like to avoid reintroducing sympy as a dependency as much as possible

Reply:
This isn't a great function. Basically I ran all these lines together in a couple of places, so I just grouped them into a function. All it is attempting to do is get the Gibbs reference energy equation into the same format whether it originated from a TDB (with nested piecewise) or a DAT (with just simple coefficients).

Piecewise equations have to be simplified as far as possible, but then I needed `expand` to get coefficients of simple powers of `T`.

Reply:
After playing around with it, I'll add that `simplified_parameter.simplify()` doesn't yield the same results as `simplify(simplified_parameter)` (and the difference results in an error).